controller/src/beerocks/master/db/db.cpp

```diff
@@ -2418,6 +2479,431 @@ const std::list<sChannelScanResults> &db::get_channel_scan_results(const sMacAdd
+        if (!update_client_entry_in_persistent_db(mac, values_map)) {
+            LOG(ERROR) << "failed to update client entry in persistent-db to for " << mac;
+            return false;
+        }
+    }
+
+    n->client_time_life_delay_sec  = time_life_delay_sec;
+    n->client_parameters_last_edit = timestamp;
+
+    return true;
+}
+
+std::chrono::seconds db::get_client_time_life_delay(const sMacAddr &mac)
+{
+    auto n = get_node(mac);
+    if (!n) {
+        LOG(ERROR) << "node not found for mac " << mac;
+        return std::chrono::seconds::zero();
+    } else if (n->get_type() != beerocks::TYPE_CLIENT) {
+        LOG(ERROR) << __FUNCTION__ << "node " << mac << " is not a valid client!";
+        return std::chrono::seconds::zero();
+    }
+
+    return n->client_time_life_delay_sec;
+}
+
+bool db::set_client_stay_on_initial_radio(const sMacAddr &mac, bool stay_on_initial_radio,
+                                          bool save_to_persistent_db)
+{
+    auto n = get_node(mac);
+    if (!n) {
+        LOG(ERROR) << "node not found for mac " << mac;
+        return false;
+    } else if (n->get_type() != beerocks::TYPE_CLIENT) {
+        LOG(ERROR) << __FUNCTION__ << "node " << mac << " is not a valid client!";
+        return false;
+    }
+
+    LOG(DEBUG) << __FUNCTION__ << ", stay_on_initial_radio = " << stay_on_initial_radio;
+
+    auto timestamp = std::chrono::steady_clock::now();
+    if (save_to_persistent_db) {
+        LOG(DEBUG) << __FUNCTION__ << ", configuring persistent-db, initial_radio_enable = "
+                   << stay_on_initial_radio;
+
+        std::unordered_map<std::string, std::string> values_map;
+        values_map.at(timestamp_str)            = timestamp_to_string_seconds(timestamp);
+        values_map.at(initial_radio_enable_str) = std::to_string(stay_on_initial_radio);
+
+        // update the persistent db
+        if (!update_client_entry_in_persistent_db(mac, values_map)) {
+            LOG(ERROR) << "failed to update client entry in persistent-db to for " << mac;
+            return false;
+        }
+    }
+
+    n->client_stay_on_initial_radio =
+        (stay_on_initial_radio) ? ePersistentParamBool::ENABLE : ePersistentParamBool::DISABLE;
+    n->client_parameters_last_edit = timestamp;
+
+    return true;
+}
+
+ePersistentParamBool db::get_client_stay_on_initial_radio(const sMacAddr &mac)
+{
+    auto n = get_node(mac);
+    if (!n) {
+        LOG(ERROR) << "node not found for mac " << mac;
+        return ePersistentParamBool::NOT_CONFIGURED;
+    } else if (n->get_type() != beerocks::TYPE_CLIENT) {
+        LOG(ERROR) << __FUNCTION__ << "node " << mac << " is not a valid client!";
+        return ePersistentParamBool::NOT_CONFIGURED;
+    }
+
+    return n->client_stay_on_initial_radio;
+}
+
+bool db::set_client_initial_radio(const sMacAddr &mac, const sMacAddr &initial_radio_mac,
+                                  bool save_to_persistent_db)
+{
+    auto n = get_node(mac);
```

Comment:
```suggestion
    auto node = get_node(mac);
```

Reply:
replaced all new-code occurrences of `n` to `node`